### PR TITLE
Remove ineffective exclusion of Arcs without parent Axes.

### PR DESCRIPTION
**Pull request by anntzer**
The `if not hasattr(self, 'axes'): raise RuntimeError(...)` check was
ineffectual, as artists now always have an Axes attribute, which can
just be None for some artists.  In fact, small Arcs are drawn just fine
without a parent Axes; e.g.
```
from pylab import *
from matplotlib.patches import *
fig = figure()
fig.add_artist(Ellipse((.2, .2), .1, .3, angle=45))  # for comparison
fig.add_artist(Arc((.2, .2), .1, .3, angle=45, theta1=0, theta2=45))
```
works just fine.  Remove the check, and adjust the docs accordingly.

On the other hand, large arcs *did* previously fail,
but that occurred a bit further down, when computing
`transforms.BboxTransformTo(self.axes.bbox)` (`self.axes` is None -->
AttributeError).  Fix that by using the figure bbox in that case (as the
point is to limit the drawing to the unclipped area, which is the whole
figure for Arcs without a parent Axes).

## PR Checklist

**Tests and Styling**
- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [ ] New features are documented, with examples if plot related.
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

File: lib/matplotlib/patches.py [hunk 1]
```diff
@@ -1997,12 +1992,11 @@ def draw(self, renderer):
...
 
-        1. The points where the ellipse intersects the axes bounding
-           box are located.  (This is done be performing an inverse
-           transformation on the axes bbox such that it is relative
-           to the unit circle -- this makes the intersection
-           calculation much easier than doing rotated ellipse
-           intersection directly).
+        1. The points where the ellipse intersects the axes (or figure)
+           bounding box are located.  (This is done be performing an inverse
+           transformation on the axes bbox such that it is relative to the unit
```
Comment (QuLogic) [inline lib/matplotlib/patches.py:1997]:
```suggestion
           bounding box are located.  (This is done by performing an inverse
           transformation on the bbox such that it is relative to the unit
```

**Review by oscargus — APPROVED** [review]

**Comment by tacaswell** [comment]
I am surprised this has no tests.

**Review by QuLogic — APPROVED** [review]

**Comment by tacaswell** [comment]
I am pretty sure that the coverage error here is a false-negative and I believe this will be further tested by #23340 .

**Comment by tacaswell** [comment]
Test failures appear to all be the font issues.

**Pull request merged**